src/sqlfluff/dialects/dialect_clickhouse.py

```diff
@@ -1059,6 +1074,85 @@ class ColumnConstraintSegment(BaseSegment):
     )
 
 
+class IndexDefinitionSegment(BaseSegment):
+    """A ClickHouse INDEX definition for CREATE TABLE.
+
+    As specified in
+    https://clickhouse.com/docs/en/sql-reference/statements/create/table#indices
+    """
+
+    type = "index_definition"
+
+    match_grammar = Sequence(
+        "INDEX",
+        Ref("SingleIdentifierGrammar"),  # index name
+        OneOf(
+            # With parentheses: INDEX name (column1, column2)
+            Bracketed(
+                Delimited(
+                    Ref("ExpressionSegment"),  # index expression
+                    delimiter=Ref("CommaSegment"),
+                )
+            ),
+            # Without parentheses: INDEX name column1
+            Ref("ExpressionSegment"),
+        ),
+        "TYPE",
+        OneOf(
+            StringParser("MINMAX", CodeSegment, type="index_type_identifier"),
```

Comment:
You don't need to invoke StringParser here, you can just use "MINMAX" to reference the minmax keyword